src/AdminConnection.ts

```diff
@@ -1,1213 +1,1218 @@
-     * @returns {Promise<any>}
-     */
-    getCompactRepository(host: string, update: boolean, timeoutMs: number): Promise<any> {
-        if (Connection.isWeb()) {
-            return Promise.reject('Allowed only in admin');
-        }
-        if (!update && this._promises.repoCompact) {
-            return this._promises.repoCompact;
-        }
-
-        if (!this.connected) {
-            return Promise.reject(NOT_CONNECTED);
-        }
-
-        if (!host.startsWith('system.host.')) {
-            host += 'system.host.' + host;
-        }
-
-        this._promises.repoCompact = new Promise((resolve, reject) => {
-            let timeout = setTimeout(() => {
-                if (timeout) {
-                    timeout = null;
-                    reject('getCompactRepository timeout');
-                }
-            }, timeoutMs || this.props.cmdTimeout);
-
-            this._socket.emit('getCompactRepository', host, data => {
-                if (timeout) {
-                    clearTimeout(timeout);
-                    timeout = null;
-                    if (data === PERMISSION_ERROR) {
-                        reject('May not read "getCompactRepository"');
-                    } else if (!data) {
-                        reject('Cannot read "getCompactRepository"');
-                    } else {
-                        resolve(data);
-                    }
-                }
-            });
-        });
-
-        return this._promises.repoCompact;
-    }
-
-    /**
-     * Get the list of all hosts in compact form (only _id, common.name, common.icon, common.color, native.hardware.networkInterfaces)
-     * @param {boolean} [update] Force update.
-     * @returns {Promise<ioBroker.Object[]>}
-     */
-    getCompactHosts(update: boolean): Promise<ioBroker.Object[]> {
-        if (Connection.isWeb()) {
-            return Promise.reject('Allowed only in admin');
-        }
-        if (!update && this._promises.hostsCompact) {
-            return this._promises.hostsCompact;
-        }
-
-        if (!this.connected) {
-            return Promise.reject(NOT_CONNECTED);
-        }
-
-        this._promises.hostsCompact = new Promise((resolve, reject) =>
-            this._socket.emit('getCompactHosts', (err, systemConfig) =>
-                err ? reject(err) : resolve(systemConfig)));
-
-        return this._promises.hostsCompact;
-    }
-}
+import { Connection, ERRORS } from "./Connection";
+import type { ConnectionProps } from "./ConnectionProps";
+import type {
+	AdminEmitEvents,
+	AdminListenEvents,
+	CompactAdapterInfo,
+	CompactHost,
+	CompactInstalledInfo,
+	CompactInstanceInfo,
+	CompactRepository,
+	Logfile,
+} from "./SocketEvents";
+import {
+	getObjectViewResultToArray,
+	normalizeHostId,
+	objectIdToHostname,
+} from "./tools";
+
+interface Certificate {
+	name: string;
+	type: "public" | "private" | "chained";
+}
+
+function parseCertificate(name: string, cert: string): Certificate | undefined {
+	if (!cert) return;
```

Comment:
IMHO in proper TypeScript, this should be `return undefined;`

Reply:
I've changed the return type to ` | void` and just return nothing. From the outside `return` and `return undefined` are equivalent though.